firstvoices/backend/views/dictionary_views.py

```diff
@@ -52,3 +57,116 @@ def get_queryset(self):
+        if selected_word is not None:
+            queryset = DictionaryEntry.objects.filter(
+                id=selected_word.dictionary_entry.id
+            )
+            if queryset.count() > 0:
+                return queryset
+
+        # Case 2. If no words found with today's date, Get words which have not yet been assigned word-of-the-day
+        words_used = WordOfTheDay.objects.filter(site__slug=site_slug).values_list(
+            "dictionary_entry_id", flat=True
+        )
+        queryset = DictionaryEntry.objects.filter(
+            site__slug=site_slug,
+            type=DictionaryEntry.TypeOfDictionaryEntry.WORD,
+            exclude_from_wotd=False,
+        ).exclude(id__in=list(words_used))
+        if queryset.count() > 0:
+            selected_word = queryset.first()
+            WordOfTheDay(
+                date=today, dictionary_entry=selected_word, site=selected_word.site
+            ).save()
+            return queryset
+
+        # Case 3. If no words found satisfying any of the above condition, try to find wotd which has not
+        # been assigned a date in the last year
+        last_year_date = today - timedelta(weeks=52)
+        selected_word = (
+            WordOfTheDay.objects.filter(site__slug=site_slug)
+            .order_by("-date")
+            .distinct()
+            .filter(date__lte=last_year_date)
+            .last()
+        )
+        if selected_word is not None:
+            queryset = DictionaryEntry.objects.filter(
+                id=selected_word.dictionary_entry.id
+            )
+            if queryset.count() > 0:
+                WordOfTheDay(
+                    date=today,
+                    dictionary_entry=selected_word.dictionary_entry,
+                    site=selected_word.dictionary_entry.site,
+                ).save()
+                return queryset
+
+        # Case 4. If there is no word that passes any of the above conditions, choose a word at random
+        primary_keys_list = DictionaryEntry.objects.filter(
+            site__slug=site_slug,
+            type=DictionaryEntry.TypeOfDictionaryEntry.WORD,
+            exclude_from_wotd=False,
+        ).values_list("id", flat=True)
+        if len(primary_keys_list) == 0:
+            # No words found
+            return DictionaryEntry.objects.none()
+        random_entry = choice(primary_keys_list)
+        selected_word = DictionaryEntry.objects.filter(id=random_entry)
+        WordOfTheDay(
+            date=today, dictionary_entry=selected_word, site=selected_word.site
+        ).save()
+        return selected_word
+
+    def list(self, request, *args, **kwargs):
+        # Overriding list method from FVPermissionViewSetMixin to only get the first word
+        site = self.get_validated_site()
+
+        # Logic to select queryset
+        if site.count() > 0:
+            queryset = self.get_suitable_queryset(site[0].slug)
+        else:
+            queryset = DictionaryEntry.objects.none()
```

Comment:
I don't think this logic is quite right-- get_validated_site will always return a site. (If it can't, it raises an error.)